### ISSUE TITLE
[Feature]: Investigate official Python extension events for env create/update/delete

### ISSUE THREAD

**Issue by etienneschalk** [etienneschalk/scientific-data-viewer#90]
### Feature Category

Accessibility

### Priority

Low - Nice to have

### Feature Description

See original issue #32 (derived from)

When manually deleting a `.venv` folder, it seems that the Python extension does not send an event, so the refresh (to find a fallback) cannot be triggered, and the state in Status Bar does not reflect anymore the reality.

This may be hard to fix is the problem comes from the official Python extension but I did not investigate yet.

This may been related to: (read below)

There is a known issue with FileWatchers though: they are not able to see deletion of directories easily. If a user manually `rm -rf .venv` for example, the change will likely not be detected and the python env must be refreshed. This is a VSCode limitation. 

## TODO 

- [ ] Dig the git history to see the unsuccessful attempt at using FileWatchers to watch env folder deletion
- [ ] Issues with the official Python extension that does not send reliably create/delete events (see function `setupOfficialPythonExtensionChangeListeners` and look for 'added/removed/updated' -> these events seem never triggered https://github.com/microsoft/vscode-python/wiki/Python-Environment-APIs)
- [ ] File system listener to detect venv deletion (low priority) to do auto reload (currently broken, only solution is manual refresh) 

 _Originally posted by @etienneschalk in [#32](https://github.com/etienneschalk/scientific-data-viewer/issues/32#issuecomment-3383487283)_

### Problem Statement

bad venv deletion event detection

### Proposed Solution

no solution for now

### Use Cases

when the user manually deletes a configured venv

### Code of Conduct

- [x] I have searched existing issues and this is not a duplicate
- [x] I have read and agree to follow the [Contributing Guidelines](CONTRIBUTING.md)
